uber/sep_commands.py

```diff
@@ -2,6 +2,88 @@
 
 
 @entry_point
+def alembic(*args):
+    """
+    Frontend for alembic script with additional uber specific facilities.
+
+    "sep alembic" supports all the same arguments as the regular "alembic"
+    command, with the addition of the "--plugin PLUGIN_NAME" option.
+
+    Passing "--plugin PLUGIN_NAME" will choose the correct alembic version path
+    for the given plugin. If "--plugin" is omitted, it will default to "uber".
+
+    If "--version-path PATH" is also specified, it will override the version
+    path chosen for the plugin. This functionality is rarely needed, and best
+    left unused.
+
+    If a new migration revision is created for a plugin that previously did
+    not have any revisions, then a new branch label is applied using the
+    plugin name. For example::
+
+        sep alembic --plugin new_plugin revision --autogenerate -m "Initial migration"
+
+    A new revision script will be created in "new_plugin/alembic/versions/"
+    with a branch label of "new_plugin". The "new_plugin/alembic/versions/"
+    directory will be created if it does not already exist.
+
+    If "--branch-label LABEL" is also specified, it will override the plugin
+    name as the new branch label.
+    """
+    argv = args if args else sys.argv[1:]
+
+    # Extract the "--plugin" option from argv.
+    plugin_name = 'uber'
+    for plugin_opt in ('-p', '--plugin'):
+        if plugin_opt in argv:
+            plugin_index = argv.index(plugin_opt)
+            argv.pop(plugin_index)
+            plugin_name = argv.pop(plugin_index)
+
+    from glob import glob
+    from os.path import exists, join
+    from alembic.config import CommandLine
+    from sideboard.config import config as sideboard_config
+    from uber.migration import create_alembic_config, version_locations
+
+    assert plugin_name in version_locations, (
+        'Plugin "{}" does not exist in {}'.format(
+            plugin_name, sideboard_config['plugins_dir']))
+
+    commandline = CommandLine(prog='sep alembic')
+    if any([h in argv for h in ('-h', '--help')]):
+        # If "--help" is passed, add a description of the "--plugin" option
+        commandline.parser.add_argument(
+            '-p', '--plugin',
+            type=str,
+            default='uber',
+            help='Plugin in which to add new versions')
```

Comment:
This will probably rarely ever be read, but I added it for completeness sake. If you tried to use a plugin name it didn't recognize, it says "Couldn't find "whatever" in /path/to/sideboard/plugins". It displays alongside the alembic help, as though it were an actual option to alembic.

The real docs can be found here: https://github.com/magfest/ubersystem/tree/fix-199-alembic-migration-branches/alembic